src/analyzer/analyzerwaveform.cpp

```diff
@@ -221,6 +247,15 @@ bool AnalyzerWaveform::processSamples(const CSAMPLE* buffer, SINT count) {
         storeIfGreater(&m_stride.m_filteredData[Left][High], chigh[Left]);
         storeIfGreater(&m_stride.m_filteredData[Right][High], chigh[Right]);
 
+        for (int s = 0; s < stemCount; s++) {
+            CSAMPLE cstem[2] = {
+                    fabs(pIn[i * stemCount + s * mixxx::kAnalysisChannels]),
+                    fabs(pIn[i * stemCount + s * mixxx::kAnalysisChannels +
+                            1])};
+            storeIfGreater(&m_stride.m_stemData[Left][s], cstem[Left]);
+            storeIfGreater(&m_stride.m_stemData[Right][s], cstem[Right]);
```

Comment:
This should become a fee function in the anonymous namespace. The whole loop should be copied to sample.cpp near:  `SampleUtil::sumAbsPerChannel` with a vectorization comment.

Reply:
Are you talking about the for-loop at line `222` of the one at `250`?

Reply:
@daschuer friendly ping in case you missed my reply here

Reply:
I original ment 250. However I have no strong opinion here. There are two overlapping principals, my comment is sourced from:
* Private functions not using members shall be moved to the anonymous namespace 
* All sample loops where vectorization matters shall go sample.cpp

Reply:
I'm afraid there may some confusion here:
- That for-loop iterates over the 4 stems, there is nothing to do about sample directly, and vectorization doesn't matter.
- `storeIfGreater` is a very specific to analyzer and IMHO it makes no sense to move it the sample helper function as there is no usecase to use it anywhere currently.
Happy to remove the for-loop and duplicate the the logic for each channel, like it is done for each frequency space above similar to this
```
            CSAMPLE cstem[2] = {
                    fabs(pIn[i * stemCount]),
                    fabs(pIn[i * stemCount + 1])};
            storeIfGreater(&m_stride.m_stemData[Left][0], cstem[Left]);
            storeIfGreater(&m_stride.m_stemData[Right][0], cstem[Right]);
            cstem[2] = {
                    fabs(pIn[i * stemCount + mixxx::kAnalysisChannels]),
                    fabs(pIn[i * stemCount + mixxx::kAnalysisChannels + 1])};
            storeIfGreater(&m_stride.m_stemData[Left][1], cstem[Left]);
            storeIfGreater(&m_stride.m_stemData[Right][1], cstem[Right]);
            cstem[2] = {
                    fabs(pIn[i * stemCount + 2 * mixxx::kAnalysisChannels]),
                    fabs(pIn[i * stemCount + 2 * mixxx::kAnalysisChannels +
                            1])};
            storeIfGreater(&m_stride.m_stemData[Left][2], cstem[Left]);
            storeIfGreater(&m_stride.m_stemData[Right][2], cstem[Right]);
            cstem[2] = {
                    fabs(pIn[i * stemCount + 3 * mixxx::kAnalysisChannels]),
                    fabs(pIn[i * stemCount + 3 * mixxx::kAnalysisChannels +
                            1])};
            storeIfGreater(&m_stride.m_stemData[Left][3], cstem[Left]);
            storeIfGreater(&m_stride.m_stemData[Right][3], cstem[Right]);
```

Reply:
Ok, than let's keep it like it is and move on.